src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonConverter.cs

```diff
@@ -154,15 +154,52 @@ public IEnumerable<(FunctionJsonSchema schema, FileInfo outputFile)?> GenerateFu
         
         private bool TryGenerateFunctionJsons()
         {
-            var assembly = Assembly.LoadFrom(_assemblyPath);
-            var functions = GenerateFunctions(assembly.GetExportedTypes()).ToList();
-            foreach (var function in functions.Where(f => f.HasValue && !f.Value.outputFile.Exists).Select(f => f.Value))
+            var assemblyDirectory = Path.GetDirectoryName(_assemblyPath);
+            if (!Directory.Exists(assemblyDirectory))
+                return false;
+
+            List<(FunctionJsonSchema schema, FileInfo outputFile)?> functions = 
+                new List<(FunctionJsonSchema, FileInfo)?>();
+
+            var assemblyFiles = Directory.GetFiles(assemblyDirectory, "*.dll", SearchOption.TopDirectoryOnly);
+            foreach (var assemblyFile in assemblyFiles)
             {
+                /// --- output message that the file is going to be scanned.
+                var _assemblyPath = Path.Combine(assemblyDirectory, assemblyFile);
+                try
+                {
+                    var assembly = Assembly.LoadFrom(_assemblyPath);
+                    var _functions = GenerateFunctions(assembly.GetExportedTypes())
+                        .ToList();
+
+                    /// --- file contained x functions
+                    if(_functions.Count!=0)
+                        functions.AddRange(_functions);
+                 
+                    _functions.Clear();
+                    _functions = null;
+                }
+                catch (Exception ex)
+                {
+                    /// --- file may not be a .net dll that we've tried to load
```

Comment:
Having an empty catch here is probably never going to be accepted. There should be a specific exception caught here. In this case, you're probably trying to catch `BadImageFormatException`, but `Assembly.LoadFrom()` can result in a number of errors. Also keep in mind that an .exe could be an assembly as well.

Reply:
@jeffputz thanks for your feedback, yes I agree with your comment with regards the try {} catch( Exception ex) {} finally {} this was a placeholder that was to capture any issues at build and echo a message/warning/error accordingly to build output.  I'm personally not interested in '.exe' files as these are not (AFAIA) used by the AzureFunction runtime currently.

Reply:
It probably doesn't matter... they committed the right change in reference to #274.